capture.go

```diff
@@ -149,18 +154,54 @@ func alert(exception error) error {
 		return true
 	})
 
+	// Run handlers in goroutines, so that if one handler is deadlocked
+	// it does not prevent others from running, or us from returning.
+	type captID struct {
+		CaptureProvider
+		CaptureID
+	}
+	captured := make(chan captID)
+	defer close(captured)
+	
+	timer := time.NewTimer(CaptureTimeout)
+	defer timer.Stop()
+	var timedOut atomic.Bool
+
+	// start a goroutine for each handler
 	for provider, handler := range capture {
-		defer func() {
-			if r := recover(); r != nil {
-				log.Printf("failed to capture exception (%q): %+v", provider, r)
+		provider := provider
+		handler := handler
+		go func() {
+			defer func() {
+				if r := recover(); r != nil {
+					log.Printf("failed to capture exception (%q): %+v", provider, r)
+				}
+			}()
+
+			id := handler(exception, arg...)
+
+			if !timedOut.Load() {
+				captured <- captID{provider, id}
```

Comment:
This can panic.
There is no guarantee that between the `Load` and the channel send that the channel hasn't been closed.

To fix it, you can use a `WaitGroup` and only close the channel when all the go-routines have completed.
You will also need to use a buffered channel so that none of the go-routines end up blocked on the send because nothing is reading from the channel anymore.

Reply:
new logic uses `done` channel and mutex, and I believe avoids the problems you've described.